src/ExceptionHandler.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <exception>
+#include <functional>
+#include <utility>  // exchange
+
+namespace opcua {
+
+class ExceptionHandler {
```

Comment:
I guess this class would be something like an `ExceptionCatcher` in this context?

Reply:
Or maybe `ExceptionPropagation`?

Reply:
Renamed to `ExceptionCatcher`, good idea! 👍